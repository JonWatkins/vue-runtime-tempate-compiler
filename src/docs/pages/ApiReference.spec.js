import { shallowMount } from '@vue/test-utils'
import ApiReference from './ApiReference.vue'
const { localVueInst } = global

let wrapper

describe('ApiReference', () => {
  beforeEach(() => {
    wrapper = shallowMount(ApiReference, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('ApiReference')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
