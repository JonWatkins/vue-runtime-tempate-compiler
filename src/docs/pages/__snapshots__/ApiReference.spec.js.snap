// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiReference render correctly 1`] = `
<div class="page">
  <h1 id="api-reference" class="display-4 mb-3">
    <router-link-stub to="/api-reference#api-reference" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      API Reference
    </router-link-stub>
  </h1>
  <h2 id="api-reference-component" class="mb-4 text-muted">
    <router-link-stub to="/api-reference#api-reference-component" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span> <code>&lt;runtime-template-compiler&gt;</code></router-link-stub>
  </h2>
  <p><code>&lt;runtime-template-compiler&gt;</code> is the component for
    enabling the compilation of templates at runtime. It takes the data
    computed method and other required properties from the parent component
    and merges them into a new one which is then merged with the results of
    the <code>vue-template-compiler</code> liraray, it is then rendered using
    the
    <a href="https://vuejs.org/" target="_blank" rel="nofollow" class="external">Vue.js<i class="fas fa-external-link-alt"></i></a> <code class="ml-1">&lt;component&gt;</code> component. This allows all of
    the normal
    <a href="https://vuejs.org/" target="_blank" rel="nofollow" class="external">Vue.js<i class="fas fa-external-link-alt"></i></a>
    bindings to work on the compiled template.
  </p>
  <h2 id="api-reference-props" class="mb-4 text-mmuted">
    <router-link-stub to="/api-reference#api-reference-props" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span> <code>&lt;runtime-template-compiler&gt;</code> Props
    </router-link-stub>
  </h2>
  <h3 id="api-reference-props-template" class="mb-4 text-muted">
    <router-link-stub to="/api-reference#api-reference-props-template" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      template
    </router-link-stub>
  </h3>
  <ul>
    <li>Type: <code>String</code></li>
    <li>Required</li>
    <li>
      The HTML template string that we would like to compile and bind data
      too.
      <pre class="language-js mb-4"><code>&lt;runtime-template-compiler :template="template" /&gt;</code></pre>
    </li>
  </ul>
  <h3 id="api-reference-props-parent" class="mb-4 text-muted">
    <router-link-stub to="/api-reference#api-reference-props-parent" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      parent
    </router-link-stub>
  </h3>
  <ul>
    <li>Type: <code>Object&lt;Component&gt;</code></li>
    <li>Optional</li>
    <li>
      The parent component we would like to fetch and bind data from.
      <pre class="language-js mb-4"><code>&lt;runtime-template-compiler :parent="parent" /&gt;</code></pre>
    </li>
  </ul>
  <h3 id="api-reference-props-compiler-options" class="mb-4 text-muted">
    <router-link-stub to="/api-reference#api-reference-props-compiler-options" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      compilerOptions
    </router-link-stub>
  </h3>
  <ul>
    <li>Type: <code>Object</code></li>
    <li>Optional</li>
    <li>
      The options to pass along to the <code>vue-template-compiler</code>.
      Check out the
      <a href="https://github.com/vuejs/vue/tree/dev/packages/vue-template-compiler#readme" target="_blank" rel="nofollow" class="external">documentation<i class="fas fa-external-link-alt"></i></a>
      for all of the avaliable options.
      <pre class="language-js mb-4"><code>&lt;runtime-template-compiler :compiler-options="options" /&gt;</code></pre>
    </li>
  </ul>
  <pagenav-stub></pagenav-stub>
</div>
`;
